chevah/compat/tests/normal/test_testing.py

```diff
@@ -22,3 +26,53 @@ def test_avatar_unicode(self):
         avatar = mk.makeFilesystemApplicationAvatar()
         self.assertIsInstance(avatar.name, text_type)
         self.assertIsInstance(avatar.home_folder_path, text_type)
+
+    def test_tearDown_excepted_threads(self):
+        """
+        Will not fail if a thread name is in excepted_threads.
+        """
+        event = Event()
+
+        class TestThread(Thread):
+            def run(self):
+                event.wait()
+
+        old_excepted_threads = self.excepted_threads[:]
+        self.excepted_threads = self.excepted_threads + ['TestThread']
+
+        thread = TestThread(name="TestThread")
+        thread.start()
+
+        self.tearDown()
+
+        self.assertTrue(thread.is_alive())
+        event.set()
+
+        # Wait for the thread to stop.
+        time.sleep(0.1)
+        self.assertFalse(thread.is_alive())
+
+        self.excepted_threads = old_excepted_threads[:]
```

Comment:
maybe use addCleanup to revert this... but I was thinking that the default testcase already has a helper which restore all instance variables after a test, so we don't need to restore them